File: tiq-e/math/2/main.py
from time import perf_counter


class Solution:
    def countPrimes(self, n: int) -> int:
        if n < 2:
            return 0
        strikes = [1] * n

        strikes[0] = 0
        strikes[1] = 0

        for i in range(2, int(n ** 0.5) + 1):
            if strikes[i] != 0:
                strikes[i * i:n:i] = [0] * ((n - 1 - i * i) // i + 1)

        return sum(strikes)


n = 10

start = perf_counter()
result = Solution().countPrimes(n)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/math/2/main.rs
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn count_primes(n: i32) -> i32 {
        if n < 3 {
            return 0;
        }
        let mut primes = n - 2;
        let mut is_prime = vec![true; n as usize];
        let mut i = 2;
        while i * i < n as usize {
            if is_prime[i] {
                let mut j = i * i;
                while j < n as usize {
                    if is_prime[j] {
                        is_prime[j] = false;
                        primes -= 1;
                    }
                    j += i;
                }
            }
            i += 1;
        }
        primes
    }
}

fn main() {
    let n = 10;

    let start = Instant::now();
    let result = Solution::count_primes(n);
    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", result);
}
